bundles/org.openhab.binding.tivo/src/main/resources/OH-INF/thing/thing-types.xml

```diff
@@ -0,0 +1,153 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<thing:thing-descriptions bindingId="tivo"
+	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xmlns:thing="https://openhab.org/schemas/thing-description/v1.0.0"
+	xsi:schemaLocation="https://openhab.org/schemas/thing-description/v1.0.0 https://openhab.org/schemas/thing-description-1.0.0.xsd">
+
+	<thing-type id="sckt">
+		<label>TiVo DVR</label>
+		<description>Monitor
+			and control your TiVo via DIRECT SOCKET commands leveraging the TIVO protocol 1.1 specification.
+			The TiVo TCP Control Protocol is an ASCII-based command protocol for remote control of a TiVo DVR
+			over a TCP local
+			network connection. The commands allow control of channel changes,
+			user interface navigation and allow the client to
+			send simulated remote control button presses to the Digital Video
+			Recorder.
+		</description>
+
+		<channels>
+			<channel id="channelSet" typeId="channelSet"/>
+			<channel id="channelForce" typeId="channelForce"/>
+			<channel id="isRecording" typeId="isRecording"/>
+			<channel id="menuTeleport" typeId="menuTeleport"/>
+			<channel id="irCommand" typeId="irCommand"/>
+			<channel id="kbdCommand" typeId="kbdCommand"/>
+			<channel id="dvrStatus" typeId="dvrStatus"/>
+		</channels>
+
+		<config-description>
+			<parameter name="address" type="text" required="true">
+				<default></default>
+				<required>true</required>
```

Comment:
The `required` tag is deprecated. You could use the attribute instead: `required="true"`